src/main/resources/templates/nonceStateCheck.html

```diff
@@ -0,0 +1,252 @@
+    const sendStateAndNonce = async () => {
+      // send state message
+      const state_uuid = await ltiPostMessage.create_UUID();
+      await ltiPostMessage.sendMessage({
+        subject: 'lti.get_data',
+        key: `state${stateHash}`,
+        value: stateHash,
+        message_id: state_uuid // an ID that we can reference on the LTI page
+      }, origin).then(function (response) {
+        console.info('State response from LMS', response);
+        stateValid = verifyAnswer(state_uuid,stateHash,response);
+        if (stateValid) {
+            document.getElementById('pmState').value = response.value;
+        }
+        verifyStateAndNonce();
+      }).catch(function (error) {
+        console.error('Error sending state message to LMS', error);
+      });
+      const nonce_uuid = await ltiPostMessage.create_UUID();
+      // send nonce message
+      await ltiPostMessage.sendMessage({
+        subject: 'lti.get_data',
+        key: `nonce${nonce}`,
+        value: nonce,
+        message_id: nonce_uuid // and ID that we can reference on the LTI page
+      }, origin).then(function (response) {
+        console.info('Nonce response from LMS', response);
+        nonceValid = verifyAnswer(nonce_uuid,nonce,response);
+        if (nonceValid) {
+            document.getElementById('pmNonce').value = response.value;
+        }
+        verifyStateAndNonce();
+      }).catch(function (error) {
+        console.error('Error sending nonce message to LMS', error);
+      });
+    }
+
+    // send state and nonce
+    sendStateAndNonce();
+    if (stateNonceValid){
+        //HERE WE WILL NEED TO FORWARD AGAIN TO OUR BACKEND!
+        var form = document.getElementById('endpoint-form');
+        //form.submit();
+    } else {
+        setAutoRedirectUrl('ltiError.html');
+    }
+
+    // Optional: Cleanup if needed when the page is unloaded
+    window.addEventListener('unload', function () {
+      // or call this method directly when postMessage service isn't needed anymore
+      ltiPostMessage.destroy();
+    });
+    }
+    if ([[${ltiStorageTarget }]]) {
+        if (window.location !== window.parent.location) {
+            nonceStateCheck();
+        } else {
+            document.getElementById('pmExpected_state').value = [[${ expected_state }]];
+            document.getElementById('pmExpected_nonce').value = [[${ expected_nonce }]];
+            document.getElementById('pmId_token').value = [[${ id_token }]];
+            document.getElementById('pmToken').value = [[${ token }]];
+            document.getElementById('pmNonce').value = [[${ expected_nonce }]];
+            document.getElementById('pmState').value = [[${ expected_state }]];
+            document.getElementById('cookies').value = 'true';
+            document.getElementById("post_button").disabled = false;
+            windows.alert('Not using post messages, just cookies because this is not an iframe. Values have just been filled in the form to pass this step');
```

Comment:
Should this be `window` instead of `windows`?

Reply:
Also should this be an alert or a `log.info`?

Reply:
I should just delete that line... it was a dev thing...  I think it is stated on another point or way. I'll check.